src/admin/migrate.rs

```diff
@@ -8,7 +8,22 @@ diesel_migrations::embed_migrations!("./migrations");
 pub struct Opts;
 
 pub fn run(_opts: Opts) -> Result<(), Error> {
+    let config = crate::Config::default();
+
+    if config.db_primary_config.read_only_mode {
+        // The service is undergoing maintenance or mitigating an outage.
+        // Exit early to ensure we don't accidentally block the release phase.
+        //
+        // Heroku will update configuration variables even if the release command fails.
+        // This will prevent deploys while in this mode, even if there are no new migrations.
+        // https://devcenter.heroku.com/articles/release-phase#release-command-failure
```

Comment:
I tried this on a personal dyno and the web dyno did *not* restart when changing the environment if the release phase fails. Are the docs wrong? Maybe we need to always return success here?

Reply:
My interpretation of that page is that the *release* will fail (so the app won't be restarted), but the configuration changes you made in the UI will not be reverted to the original values.